src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvoker.cs

```diff
@@ -626,11 +632,23 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                         goto case State.ActionBegin;
                     }
 
-                case State.ExceptionShortCircuit:
+                case State.ExceptionHandled:
                     {
+                        // We arrive in this state when an exception happened, but was handled by exception filters
+                        // either by setting ExceptionHandled, or nulling out the Exception or setting a result
+                        // on the ExceptionContext.
+                        //
+                        // We need to execute the result (if any) and then exit gracefully which unwinding Resource 
+                        // filters.
+
                         Debug.Assert(state != null);
                         Debug.Assert(_exceptionContext != null);
 
+                        if (_exceptionContext.Result == null)
+                        {
+                            _exceptionContext.Result = new EmptyResult();
```

Comment:
We can remove the Debug.Assert on Line 655 since this ensures Result is never null.